tests/rt.sh

```diff
@@ -30,77 +30,405 @@ usage() {
+  [[ $skip_check_results == true ]] && echo "* (-w) - SKIP RESULTS CHECK" >> "${REGRESSIONTEST_LOG}"
+  [[ $KEEP_RUNDIR == true ]] && echo "* (-k) - KEEP RUN DIRECTORY" >> "${REGRESSIONTEST_LOG}"
+  [[ $ROCOTO == true ]] && echo "* (-r) - USE ROCOTO" >> "${REGRESSIONTEST_LOG}"
+  [[ $ECFLOW == true ]] && echo "* (-e) - USE ECFLOW" >> "${REGRESSIONTEST_LOG}"
+
+
+  [[ -f "${TEST_CHANGES_LOG}" ]] && rm ${TEST_CHANGES_LOG}
+  touch ${TEST_CHANGES_LOG}
+  while read -r line || [ "$line" ]; do
+    line="${line#"${line%%[![:space:]]*}"}"
+    [[ -n "$line" ]] || continue
+    [[ ${#line} == 0 ]] && continue
+    [[ $line == \#* ]] && continue
+    local valid_compile=false
+    local valid_test=false
+
+    if [[ $line == COMPILE* ]] ; then
+      
+      CMACHINES=$(echo "$line" | cut -d'|' -f5 | sed -e 's/^ *//' -e 's/ *$//')
+      COMPILER=$(echo "$line" | cut -d'|' -f3 | sed -e 's/^ *//' -e 's/ *$//')
+      COMPILE_NAME=$(echo "$line" | cut -d'|' -f2 | sed -e 's/^ *//' -e 's/ *$//')
+      COMPILE_ID=${COMPILE_NAME}_${COMPILER}
+      
+      if [[ ${CMACHINES} == '' ]]; then
+        valid_compile=true
+      elif [[ ${CMACHINES} == -* ]]; then
+        [[ ${CMACHINES} =~ ${MACHINE_ID} ]] || valid_compile=true
+      elif [[ ${CMACHINES} == +* ]]; then
+        [[ ${CMACHINES} =~ ${MACHINE_ID} ]] && valid_compile=true
+      fi
+
+      if [[ $valid_compile == true ]]; then
+        COMPILE_COUNTER=$((COMPILE_COUNTER+1))
+        FAIL_LOG=""
+        COMPILE_RESULT=""
+        TIME_FILE=""
+        COMPILE_TIME=""
+        RT_COMPILE_TIME=""
+        if [[ ! -f "${LOG_DIR}/compile_${COMPILE_ID}.log" ]]; then
+          COMPILE_RESULT="MISSING"
+          FAIL_LOG="N/A"
+        elif [[ -f fail_compile_${COMPILE_ID} ]]; then
+          COMPILE_RESULT="FAIL TO RUN"
+          FAIL_LOG="${LOG_DIR}/compile_${COMPILE_ID}.log"
+        else 
+          if grep -q "quota" "${LOG_DIR}/compile_${COMPILE_ID}.log"; then
+            COMPILE_RESULT="FAIL FROM DISK QUOTA"
+            FAIL_LOG="${LOG_DIR}/compile_${COMPILE_ID}.log"
+          elif grep -q "timeout" "${LOG_DIR}/compile_${COMPILE_ID}.log"; then
+            COMPILE_RESULT="FAIL FROM TIMEOUT"
+            FAIL_LOG="${LOG_DIR}/compile_${COMPILE_ID}.log"
+          else
+            COMPILE_RESULT="PASS"
+            TIME_FILE="${LOG_DIR}/compile_${COMPILE_ID}_timestamp.txt"
+            if [[ -f "${TIME_FILE}" ]]; then
+              while read -r times || [ "$times" ]; do
+                  times="${times#"${times%%[![:space:]]*}"}"
+
+                  DATE1=$(echo "$times" | cut -d ',' -f2 )
+                  DATE2=$(echo "$times" | cut -d ',' -f3 )
+                  DATE3=$(echo "$times" | cut -d ',' -f4 )
+                  DATE4=$(echo "$times" | cut -d ',' -f5 )
+
+                  COMPILE_TIME=$(date --date=@$((DATE3 - DATE2)) +'%M:%S')
+                  RT_COMPILE_TIME=$(date --date=@$((DATE4 - DATE1)) +'%M:%S')
+
+              done < "$TIME_FILE"
+            fi
```

Comment:
@DusanJovic-NOAA I think you did work to create the timestamp files. I'm not sure I understand what time is for what in there. Could you double check I have it right, or what corrections I might need?